language/move-vm/runtime/src/interpreter.rs

```diff
@@ -148,6 +148,17 @@ impl Interpreter {
         ty_args: Vec<Type>,
         args: Vec<Value>,
     ) -> VMResult<Vec<Value>> {
+        // A counter for tracking numbers of struct that don't have any capability.
+        // This counter will be incremented whenever a struct that has empty capability was created by pack and decremented.
+        //
+        // The invariant here is this counter should always goes to zero at the end of execution.
+        // The reason is that values with such capability can only be created by pack instruction and destructed by unpack
+        // instruction and can never be stored in storage.
+        //
+        // Thus if the counter is not zero towards the end of execution, it means that some hot potato values are silently dropped at runtime.
+
+        let mut hot_potato_counter: u64 = 0;
```

Comment:
Its fine that we do just this for now, but a more central problem of hot potatoes -- at least in the mind of the Move team when this feature was developed -- is store, not drop. Anyway good to see this.

Reply:
Key and copy can both break the invariant here: struct has key can be stored directly into storage, struct has copy can be copied and thus causing more unpack than pack.